### PR TITLE
nitrokey-fido2-firmware: init at 2.4.1

### PR DESCRIPTION
**Pull request by kiike**
## Description of changes
This is the firmware for the Nitrokey FIDO2 key, the packaging of which was continued during the Summer of Nix 2024 with sponsorship by NLNET.

This is not an executable, but a binary firmware that can be flashed onto such keys.  According to a discourse thread, inclusion of such artifacts in nixpkgs [shouldn't be an issue](https://discourse.nixos.org/t/inclusion-of-device-firmwares-in-nixpkgs/50980).

cc @amerinor01 @jleightcap @imadnyc @RCoeurjoly @fricklerhandwerk 

Related PRs:

- #337966
- #337959
- #337956
- #337954
- #337951

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by imadnyc** [comment]
@sbruder I can't believe we didn't include you earlier! Could you message me @dre:imad.nyc on matrix or email me@imad.nyc for the rest of the groups info? I can't seem to find it anywhere on the internet.

**Comment by RCoeurjoly** [comment]
> Hey, [I initially packaged the nitrokey firmware](https://github.com/ngi-nix/nitrokey-firmware/commits?author=sbruder) as part of Summer of Nix 2022. It’s great to see it now being included into nixpkgs. Could you please include me as a co-author of the commits of this and the related PRs that are based on the previous work? That would be great to indicate the provenance of the code, as there is nothing right now that links the new PRs to the initial work from 2022. However, I don’t have the time to continue working on this, so I can’t be a maintainer for the packages.

Thanks for you work! Done, you are included as a co-author. As @imadnyc said, if there was someone else involved, please let us know so that we can include them.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 337951` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/by-name/ni/nitrokey-fido2-firmware/package.nix [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    homepage = "https://github.com/Nitrokey/nitrokey-fido2-firmware";
+    maintainers = with lib.maintainers; [
+      amerino
+      kiike
+      imadnyc
+    ];
+    license = with lib.licenses; [
+      asl20
+      mit
+    ];
```
Comment (wegank) [inline pkgs/by-name/ni/nitrokey-fido2-firmware/package.nix:81]:
```suggestion
    ];
    platforms = lib.platforms.unix;
```

File: pkgs/by-name/ni/nitrokey-fido2-firmware/package.nix [hunk 2]
```diff
@@ -0,0 +1,83 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  writeShellScriptBin,
+  gcc-arm-embedded,
```
Comment (wegank) [inline pkgs/by-name/ni/nitrokey-fido2-firmware/package.nix:6]: Does `gcc-arm-embedded-13` work?

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 337951` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nitrokey-fido2-firmware</li>
  </ul>
</details>

**Pull request merged**